inference-engine/tests/functional/plugin/cpu/subgraph_tests/src/concat_conv_sum_inplace.cpp

```diff
@@ -0,0 +1,83 @@
+
+using namespace CPUTestUtils;
+using namespace InferenceEngine;
+
+namespace SubgraphTestsDefinitions {
+// Subgraph:
+/*
+ *          paramter1         parameter2
+ *                  \             /
+ *                   \           /
+ *                   ReLu1    ReLu2
+ *                   / \       /
+ *                  /   \     /
+ *                 /     \   /
+ *                 |     Concat (inPlace)
+ *                 |       |
+ *                 |      Conv
+ *                 |       |
+ *                 |      Add_Const (Bias)
+ *                  \      |
+ *                   \     |
+ *                    \    |
+ *                     \   |
+ *                       Sum
+ *                        |
+ *                       ReLu3
+ *                        |
+ *                      Result
+ */
+
+class ConcatConvSumInPlaceTest : virtual public LayerTestsUtils::LayerTestsCommon {
+public:
+    void SetUp() override {
+        const std::vector<size_t> inputShape = {1, 64, 12, 12};
+        const InferenceEngine::SizeVector kernel = {1, 1};
+        const InferenceEngine::SizeVector stride = {1, 1};
+        const InferenceEngine::SizeVector dilation = {1, 1};
+        const std::vector<ptrdiff_t> padBegin = {0, 0};
+        const std::vector<ptrdiff_t> padEnd = {0, 0};
+        const size_t convOutChannels = 64;
+        const auto targetFormat = with_cpu_x86_avx512_core() ? nChw16c : nChw8c;
+
+
+        auto inputParams = ngraph::builder::makeParams(ngraph::element::f32, {inputShape, inputShape});
+
+        auto Relu1 = std::make_shared<ngraph::opset3::Relu>(inputParams[0]);
+        Relu1->get_rt_info() = CPUTestsBase::makeCPUInfo({targetFormat}, {targetFormat}, {});
+        auto Relu2 = std::make_shared<ngraph::opset3::Relu>(inputParams[1]);
+        Relu2->get_rt_info() = CPUTestsBase::makeCPUInfo({targetFormat}, {targetFormat}, {});
+
+        auto concat = ngraph::builder::makeConcat(ngraph::OutputVector{Relu1, Relu2}, 1);
+
+        auto conv = ngraph::builder::makeConvolution(concat, ngraph::element::f32, kernel, stride, padBegin,
+                                                     padEnd, dilation, ngraph::op::PadType::AUTO, convOutChannels);
+        auto bias = ngraph::builder::makeConstant<float>(ngraph::element::Type_t::f32, ngraph::Shape({1, convOutChannels, 1, 1}), {}, true);
+        auto convBiasAdd = std::make_shared<ngraph::opset3::Add>(conv, bias);
+
+        auto sum = std::make_shared<ngraph::opset3::Add>(convBiasAdd, Relu1);
+
+        auto Relu3 = std::make_shared<ngraph::opset3::Relu>(sum);
+
+        ngraph::ResultVector results{std::make_shared<ngraph::opset3::Result>(Relu3)};
+        function = std::make_shared<ngraph::Function>(results, inputParams, "ConcatConvSumInPlace");
+        targetDevice = CommonTestUtils::DEVICE_CPU;
+    }
+};
+
+namespace {
+    TEST_F(ConcatConvSumInPlaceTest, smoke_ConcatConvSumInPlace_CPU) {
+        SKIP_IF_CURRENT_TEST_IS_DISABLED()
+
+        Run();
```

Comment:
shouldn't we also check that ```Reorder``` was inserted between ```Relu``` and ```Convolution```?

Reply:
In general case it does not matter where exactly the reorder is inserted, or whether it is inserted at all. What does matter is that the memory is actually copied. If it is not copied, then the test will fail.

Reply:
ok, let's leave it as is